app/auth/signup/page.tsx

```diff
@@ -1,5 +1,59 @@
-import { Heading } from '@chakra-ui/react'
+'use client'
+import { startTransition } from 'react'
+import { useForm } from 'react-hook-form'
+import { Flex, Heading } from '@chakra-ui/react'
+import * as z from 'zod'
+import { PrimaryButton } from '@/components/button'
+import { signUp } from './action'
+
+const signUpSchema = z.object({
+  email: z.string().email(),
+  password: z.string().min(8)
+})
+export type SignUpSchema = z.infer<typeof signUpSchema>
 
 export default function SignUp() {
-  return <Heading>SignUp Page</Heading>
+  const {
+    register,
+    handleSubmit,
+    formState: { errors }
+  } = useForm<SignUpSchema>()
+  return (
+    <>
+      <Heading>SignUp Page</Heading>
+      <form
+        onSubmit={handleSubmit((data) => {
+          const result = signUpSchema.safeParse(data)
+          if (!result.success) {
+            console.error(result.error)
+            return
+          }
+          startTransition(() => {
+            signUp(result.data).catch(console.error)
+          })
+        })}
+      >
+        <Flex flexDirection={'column'}>
+          <label htmlFor="email">Email</label>
+          <input type="email" id="email" required {...register('email')} />
+          <label htmlFor="password">Password</label>
+          <input
+            type="password"
+            id="password"
+            required
+            {...register('password', {
+              minLength: {
+                value: 8,
+                message: 'min length is 8'
+              }
+            })}
+          />
+          {errors.password && (
+            <span role="alert">{errors.password.message}</span>
+          )}
```

Comment:
How to strict email policy and password policy ?
Just as an example, added minlength policy to password.